### ISSUE TITLE
TheCombine needs to work without e-mail services

### ISSUE THREAD

**Issue by jmgrady** [sillsdev/TheCombine#509]
When TheCombine is deployed on a NUC, the internet will not usually be available so there will be no SMTP server that is reachable.  A configuration item needs to be added to indicate that e-mail is not available.  When that configuration item is set, then:

1. `BackendFramework.Startup()` should not throw any errors if the SMTP environment variables are not set.
1. The button to invite a person to join a project should not be displayed or replaced with an "Register User" button (or the like).
1. Similarly, the Forgot Password functionality will have to be disabled or replaced with a mechanism for an admin user to be able to reset a user's password.

Note that we will probably want a configuration item rather than decide based on a missing environment variable.  This will allow us to distinguish between a device that is intended to be used in a remote scenario and a configuration error.

**Comment by jmgrady** [comment]
Actually, the configuration item (in the RuntimeConfig type) will be necessary to get the UI to behave nicely since it has no access to the environment variables - unless the Backend adds a method to its API to query whether e-mail is available.  In addition, the configuration item is not available to the Backend so an additional mechanism is needed to indicate to the Backend whether e-mail services should be available, perhaps ASPNETCORE_SMTP_SERVER=none.

That said, having two independent methods for specifying whether e-mail services are available can be problematic - unless a build/installation process is created so that these are both driven by a single element in the source.

**Comment by johnthagen** [comment]
The immediate issue of the backend not starting up without `SMTP_PORT` environment variable present will be fixed in #522.

**Comment by johnthagen** [comment]
> Actually, the configuration item (in the RuntimeConfig type) will be necessary to get the UI to behave nicely since it has no access to the environment variables

Yes, we'll likely need a new `'runtimeConfig'` option for this, so the frontend knows if email services are available.

The backend should be able to just check for the existence of all `COMBINE_SMTP_*` env variables as a way to know whether email should be supported.

> That said, having two independent methods for specifying whether e-mail services are available can be problematic - unless a build/installation process is created so that these are both driven by a single element in the source.

Yeah, I'm imaging for production, Ansible will generate the correct corresponding build ARGs and ENV variables based on the deployment scenario (NUC, vs not-NUC, etc).

**Comment by imnasnainaec** [comment]
Related: #394

**Comment by jmgrady** [comment]
Testing on version 0.7.0 shows:
- when I try to invite someone by email (with no internet available), the UI shows a perpetual busy signal.  I can get out of it by selecting something else (other than the project settings).
- when I click "Forgot Password", enter my email and click _SUBMIT_, the UI says _DONE_ even though nothing happened.
